### ISSUE TITLE
Using a subspec that only has header files will also incorporate source files if another subspec in the podpsec is referenced.

### ISSUE THREAD

**Issue by jcollas** [CocoaPods/CocoaPods#735]
I have a podspec that has two subspecs, 'Headers' and 'Lib'.

Headers only references the *.h files in my library, and Lib references the *.m files, with a dependency on Headers.

If my Podfile references the Headers subspec in any of the targets, things work as expected (I don't see any *.m in the 'Compile Sources' phase of the pod target.  If there's a reference to the Lib subspec anywhere else in the Podfile (like a totally different target), then I will see the *.m files in the 'Compile Sources' phase of the pod target, even though that target does not reference anything but the 'Header' target.

I'm trying to separate out the header files from the souce files because I have a workspace of dependent projects, and in the library projects I want to refer to the header files, and in the application project, I want to link the library projects and not get duplicate symbols because I've referenced the pod in the app and library projects.

**Comment by fan2** [comment]
Just specify `source_files` same as `header_files` will work: 

```
    s.subspec 'Header' do |ss|
        ss.public_header_files = 'include/*.{h,hpp}', 'util/include/*.{h,hpp}'
        ss.source_files = 'include/*.{h,hpp}', 'util/include/*.{h,hpp}'
    end
```

**Comment by jcollas** [comment]
Thanks.  I don't even remember what pods were causing me the problem, I think it was resolved in 0.17.
